src/graphnet/models/detector/prometheus.py

```diff
@@ -38,5 +38,302 @@ def _t(self, x: torch.tensor) -> torch.tensor:
+
+    def _t(self, x: torch.tensor) -> torch.tensor:
+        return x / 1.05e04
+
+
+class ORCA150(Detector):
+    """`Detector` class for Prometheus ORCA150."""
+
+    geometry_table_path = os.path.join(
+        PROMETHEUS_GEOMETRY_TABLE_DIR, "orca.parquet"
+    )
+    xyz = ["sensor_pos_x", "sensor_pos_y", "sensor_pos_z"]
+    string_id_column = "sensor_string_id"
+    sensor_id_column = "sensor_id"
+
+    def feature_map(self) -> Dict[str, Callable]:
+        """Map standardization functions to each dimension."""
+        feature_map = {
+            "sensor_pos_x": self._sensor_pos_xy,
+            "sensor_pos_y": self._sensor_pos_xy,
+            "sensor_pos_z": self._sensor_pos_z,
+            "t": self._t,
+        }
+        return feature_map
+
+    def _sensor_pos_xy(self, x: torch.tensor) -> torch.tensor:
+        return x / 10
+
+    def _sensor_pos_z(self, x: torch.tensor) -> torch.tensor:
+        return x / 100
+
+    def _t(self, x: torch.tensor) -> torch.tensor:
+        return x / 1.05e04
+
+
+class IceCube86(Detector):
+    """`Detector` class for Prometheus IceCube86."""
+
+    geometry_table_path = os.path.join(
+        PROMETHEUS_GEOMETRY_TABLE_DIR, "icecube86.parquet"
+    )
+    xyz = ["sensor_pos_x", "sensor_pos_y", "sensor_pos_z"]
+    string_id_column = "sensor_string_id"
+    sensor_id_column = "sensor_id"
+
+    def feature_map(self) -> Dict[str, Callable]:
+        """Map standardization functions to each dimension."""
+        feature_map = {
+            "sensor_pos_x": self._sensor_pos_xy,
+            "sensor_pos_y": self._sensor_pos_xy,
+            "sensor_pos_z": self._sensor_pos_z,
+            "t": self._t,
+        }
+        return feature_map
+
+    def _sensor_pos_xy(self, x: torch.tensor) -> torch.tensor:
+        return x / 100
+
+    def _sensor_pos_z(self, x: torch.tensor) -> torch.tensor:
+        return x / 1000
+
+    def _t(self, x: torch.tensor) -> torch.tensor:
+        return x / 1.05e04
+
+
+class IceCubeDeepCore8(Detector):
+    """`Detector` class for Prometheus IceCubeDeepCore8."""
+
+    geometry_table_path = os.path.join(
+        PROMETHEUS_GEOMETRY_TABLE_DIR, "icecube_deepcore.parquet"
+    )
+    xyz = ["sensor_pos_x", "sensor_pos_y", "sensor_pos_z"]
+    string_id_column = "sensor_string_id"
+    sensor_id_column = "sensor_id"
+
+    def feature_map(self) -> Dict[str, Callable]:
+        """Map standardization functions to each dimension."""
+        feature_map = {
+            "sensor_pos_x": self._sensor_pos_xy,
+            "sensor_pos_y": self._sensor_pos_xy,
+            "sensor_pos_z": self._sensor_pos_z,
+            "t": self._t,
+        }
+        return feature_map
+
+    def _sensor_pos_xy(self, x: torch.tensor) -> torch.tensor:
+        return x / 100
+
+    def _sensor_pos_z(self, x: torch.tensor) -> torch.tensor:
+        return x / 1000
+
+    def _t(self, x: torch.tensor) -> torch.tensor:
+        return x / 1.05e04
+
+
+class IceCubeGen2(Detector):
```

Comment:
Am i understanding correctly that in this configuration we are "simply" using the features available in both the old and new detector modules? and excluding stuff like directionality of the specific PMT? Which would also require us to be able to handle heterogeneous graphs [https://pytorch-geometric.readthedocs.io/en/latest/tutorial/heterogeneous.html]

Reply:
Hey! Simulation from Prometheus "stops" at the module-level, so it does not simulate DOMs etc. All we have are photons. We don't know where on the module, although I think Prometheus could perhaps add that information in the output if configured to do so. But the simulation we're expecting will not have that.